crates/sui-indexer/src/handlers/checkpoint_handler.rs

```diff
@@ -588,59 +554,40 @@ where
             .collect::<Vec<_>>();
         let indexed_deleted_objects: Vec<IndexedDeletedObject> = deleted_objects
             .into_iter()
-            .map(|o| IndexedDeletedObject {
-                object_id: o.0,
-                object_version: o.1.value(),
+            .map(|obj_ref| IndexedDeletedObject {
+                object_id: obj_ref.0,
+                object_version: obj_ref.1.into(),
                 checkpoint_sequence_number: checkpoint_seq,
             })
             .collect();
 
-        let (latest_objects, _) = get_latest_objects(data.output_objects());
-        let history_object_map = data
-            .output_objects()
+        let latest_live_output_objects = data.latest_live_output_objects();
+        let latest_live_output_object_map = latest_live_output_objects
+            .clone()
             .into_iter()
-            .map(|o| ((o.id(), o.version()), o.clone()))
+            .map(|o| (o.id(), o.clone()))
             .collect::<HashMap<_, _>>();
 
-        let history_objects: Vec<Object> = data
+        let output_objects = data
             .transactions
             .iter()
-            .flat_map(|tx| {
-                let CheckpointTransaction {
-                    transaction: tx,
-                    effects: fx,
-                    ..
-                } = tx;
-                fx.all_changed_objects()
-                    .into_iter()
-                    .map(|(oref, _owner, _kind)| {
-                        let history_object = history_object_map.get(&(oref.0, oref.1)).unwrap_or_else(|| {
-                            panic!(
-                                "object {:?} version {:?} not found in CheckpointData (tx_digest: {})",
-                                oref.0,
-                                oref.1,
-                                tx.digest()
-                            )
-                        });
-                        assert_eq!(oref.2, history_object.digest());
-                        history_object.clone()
-                    })
-                    .collect::<Vec<_>>()
-            })
-            .collect();
-
+            .flat_map(|tx| &tx.output_objects)
+            .collect::<Vec<_>>();
         let move_struct_layout_map =
-            get_move_struct_layout_map(&history_objects, package_resolver).await?;
-        let changed_objects = history_objects
+            get_move_struct_layout_map(output_objects.clone(), package_resolver).await?;
+        let changed_objects = output_objects
             .into_iter()
             .map(|o| {
-                let df_info =
-                    try_create_dynamic_field_info(&o, &move_struct_layout_map, &latest_objects);
-                df_info.map(|info| IndexedObject::from_object(checkpoint_seq, o, info))
+                let df_info = try_create_dynamic_field_info(
+                    o,
+                    &move_struct_layout_map,
+                    &latest_live_output_object_map,
```

Comment:
This is still not correct, right? It's not using the state at the time of object mutation

Reply:
@lxfind indeed, I was aware of that but assuming that we will remove df_* fields except df_kind in a following pr, so I added a TODO above and minimize the changes here just to compile
```
// TODO(gegaowp): move_struct_layout_map is no longer needed when we only keep df_kind,
// this has been discussed and will do it in a follow-up PR with other related schema changes.
```